src/Debugger/Test/DebugSessionExtensions.cs

```diff
@@ -23,7 +23,7 @@ internal static class DebugSessionExtensions {
             EventHandler<DebugBrowseEventArgs> handler = null;
             handler = (sender, e) => {
                 session.Browse -= handler;
-                browseTcs.SetResult(true);
+                browseTcs.TrySetResult(true);
```

Comment:
Do you still need this ad-hoc or you can replace it with `EventTaskSources.DebugSession.Browse`?